8546 Connect Order to Motion to test 1761846877

zacharybrown204 opened the pull request:
For #8546 
Adding new fields to the file court issued docket entry view. 
- Adds a checkbox only for orders
- - When checked it will display pairs of dropdowns to select an order and a disposition

Added validation for above fields. 
- Note that our code processing validation errors was condensing any errors for arrays of complex objects to show as a single error for any object in that array. For example if I have an array like [{a: 1, b: 2, c:3}, {a:1, b:2, c:'error value'}], the validation errors would only say that "c" as an error and not "c" at index 1. 
We updated that validation to now report that "array-1-c" has an error. The validation errors are now returning the full path of the error so that we can pinpoint the exact spot the error is occuring in the form. This will allow us to have more accurate validation on dynamic forms. 

Added new table to hold relationship between docket entries. We tried to make it more generic in the case that there are any more relationships between docket entries in the future.

zacharybrown204 committed:
8546 Adding order columns to docket entry

JayFlexy committed:
8546: wip add comments for points of interests

JayFlexy committed:
8546: wip more points of interest

JayFlexy committed:
8546: wip add checkbox

JayFlexy committed:
Merge branch 'staging' into 8546-connect-order-to-motion

zacharybrown204 committed:
8546 - WIP API updates to add dispositions to motions

JayFlexy committed:
8546: connect checkbox

zacharybrown204 committed:
8546 WIP - API side updates

Added DB updates to file and serve interactor
Added updates to getCasesByDocketNumbers.

zacharybrown204 committed:
8546 - Adding typescript types to document metadata for the file interactors

JayFlexy committed:
8546: add new motion disposition option

JayFlexy committed:
8546: wip fix data connections

JayFlexy committed:
8546: wip debug order modal failing to confirm

zacharybrown204 committed:
8546 - Motions + Orders Connecting UI to API

JayFlexy committed:
8546: add Dispotion to motion, filter previous disposed motions

zacharybrown204 committed:
8546 - Now able to add multiple motions to the order

JayFlexy committed:
8546: branch reconstruction wip

zacharybrown204 committed:
8546

Tweaks to adding multiple motions to order

JayFlexy committed:
8546: robustify index grabbing function to prevent faulty indices

JayFlexy committed:
Merge branch 'staging' into 8546-connect-order-to-motion-revamped

zacharybrown204 committed:
8546 - small style update

JayFlexy committed:
8546: All caps dispositions

zacharybrown204 committed:
Merge branch '8546-connect-order-to-motion-revamped' of https://github.com/ustaxcourt/ef-cms into 8546-connect-order-to-motion-revamped

zacharybrown204 committed:
8546

Updating file & file and serve to account for cosolidated cases.

JayFlexy committed:
8546: add ability to hold docket entry relationships

JayFlexy committed:
8546: some comment and log cleanup

zacharybrown204 committed:
8546 Docket entry relations

Changing docket entry relation structure to be more generic.
Allowing for order documents to point back to motions as well.

zacharybrown204 committed:
8546 Adding line breaks when listing related docket entries

zacharybrown204 committed:
8546 - Updating table and column names

zacharybrown204 committed:
8546 - Added form logic when adding motions to order

Motions already selected will now be excluded from other dropdowns when associating with orders

JayFlexy committed:
rm old log statement

JayFlexy committed:
8546: fix link to child cases

JayFlexy committed:
8546: wip

zacharybrown204 committed:
8546 - Refacotring related docket entries table

Consolidating docket number columns into single column

zacharybrown204 committed:
Merge branch '8546-connect-order-to-motion-revamped' of https://github.com/ustaxcourt/ef-cms into 8546-connect-order-to-motion-revamped

JayFlexy committed:
8546: wip, start validation work on new branch

zacharybrown204 committed:
8546 - Updating edit court issued document view to handle motions + orders

JayFlexy committed:
Merge branch '8546-connect-order-to-motion-revamped' into 8546-connect-order-to-motion-revamped-validation-work

JayFlexy committed:
8546: add validation for court issued docket entry (order)

JayFlexy committed:
8546: rm outdated comment

zacharybrown204 committed:
8546 - Adding validation for motions & orders form

zacharybrown204 committed:
8546 - Add validation to edit form

zacharybrown204 committed:
8546 - Updating types in file interactor

zacharybrown204 committed:
8546 Typescript updates

JayFlexy committed:
8546: wip cypress test

JayFlexy committed:
8546: wip cypress tests

JayFlexy committed:
8546: rm CourtIssueDocumentBase

zacharybrown204 committed:
8546 - Typescript updates - fixed a circular dependency

JayFlexy committed:
Merge branch 'staging' into 8546-connect-order-to-motion-revamped

zacharybrown204 committed:
8546 - unit test updates & typescript fixes

zacharybrown204 committed:
8546 - Adding happy path tests for relating documents

JayFlexy committed:
8546: wip cypress debugging

JayFlexy committed:
8546: cypress test - order affecting a docket entry

zacharybrown204 committed:
8546 - Re-adding court issued document base as partial document

zacharybrown204 committed:
8546 - Fixing failing tests in github actions

zacharybrown204 committed:
8546 - Fixes for validation updates in shared tests

zacharybrown204 committed:
8546 - Fixing integration and cypress tests

zacharybrown204 committed:
8546 - Fixing typescript errors and unit tests.

zacharybrown204 committed:
8546 - Fixing cypress test and adding more test coverage

JayFlexy committed:
Merge branch 'staging' into 8546-connect-order-to-motion-revamped

zacharybrown204 committed:
8546 - Few DB updates based off PR feedback

JayFlexy committed:
8546: rm unused check

JayFlexy committed:
Merge branch 'staging' into 8546-connect-order-to-motion-revamped

JayFlexy committed:
8546: comment cleanup

zacharybrown204 committed:
Merge remote-tracking branch 'origin/8546-connect-order-to-motion-revamped' into 8546-connect-order-to-motion-revamped-to-test-1761846877

zacharybrown204 committed:
8546 Fixing bad merge resolution

zacharybrown204 committed:
Fixing assert on cypress test

akuny commented on shared/src/business/entities/JoiValidationEntity.ts:
```diff
@@ -104,7 +105,9 @@ export abstract class JoiValidationEntity {
     error.details.forEach(detail => {
-      if (!Number.isInteger(detail.context.key)) {
+      if (detail.path.length > 1) {
+        errors[_.join(detail.path, '-')] = detail.message;
```
Extreme nitpick: we tend to import lodash utility functions by name specifically (see line 2 in this file).

zacharybrown204 replied:
Duely noted, fixed

akuny commented on web-api/src/business/useCases/courtIssuedDocument/fileAndServeCourtIssuedDocumentInteractor.consolidatedCases.test.ts:
```diff
@@ -265,4 +278,63 @@ describe('consolidated cases', () => {
+      ],
+    };
+
+    MOCK_LEAD_CASE_WITH_PAPER_SERVICE.docketEntries.push(mockOrderDocketEntry);
```
Do we want to mutate test fixture data as such? I think we could achieve the same thing with `leadCaseDocketEntries.push(mockOrderDocketEntry);` since `leadCaseDocketEntries` is created fresh each test run: that way we aren't touching the fixture data, and there's no risk of test order dependency.

zacharybrown204 replied:
Fixed, turns out this test didn't even need that mock docket entry pushed.
I could have sworn I copied that line from another test, but I couldn't find where it was to check if it was a problem there too. Might've just picked a bad replacement for a missing reference on the copied line or something.

akuny commented on web-api/src/business/useCases/docketEntry/fileCourtIssuedDocketEntryInteractor.ts:
```diff
@@ -176,6 +176,28 @@ export const fileCourtIssuedDocketEntry = async (
     }),
   );
 
+  if (documentMeta.affectedDocketEntries) {
```
The code wrapped by this if statement is very similar to code in `fileAndServeCourtIssuedDocumentInteractor`: this could be extracted and placed in the use case helpers layer to reduce duplication.

zacharybrown204 replied:
Extracted code out to a helper function and used that instead. I know we have a bunch of helper functions going through app context and that function to get helpers, but I also know we want to get away from app context so I had just direct imported the helper.

akuny replied:
Thanks for doing that--direct importing is the way to go.

akuny commented on web-api/src/business/useCases/docketEntry/updateDocketEntryMetaInteractor.ts:
```diff
@@ -61,6 +64,16 @@ export const updateDocketEntryMeta = async (
     throw new Error('Unable to update unserved docket entry.');
   }
 
+  if (
```
Is it worth updating the accompanying unit test suite to cover this path?

zacharybrown204 replied:
Added 3 new tests to cover updating docket entries with affectedDocketEntries

akuny commented on web-client/src/presenter/computeds/Public/publicCaseDetailHelper.ts:
```diff
@@ -77,7 +77,7 @@ export const formatDocketEntryOnDocketRecord = (
   entry.filingsAndProceedings = getFilingsAndProceedings(entry);
 
-  const canPublicUserSeeLink = DocketEntry.isDownloadable(entry, {
+  const canPublicUserSeeLink = DocketEntry.isDownloadable(entry, { // NOTE (#8546): may use similar to determine if link should be active for relations
```
Are these comments still necessary?

zacharybrown204 replied:
Comment Removed

akuny commented on web-client/src/views/CourtIssuedDocketEntry/CourtIssuedDocketEntry.tsx:
```diff
@@ -154,6 +157,177 @@ export const CourtIssuedDocketEntry = connect(
+                    </fieldset>
+                  </FormGroup>
+                )}
+                {/* TODO (#8546): Fix this */}
```
Is there still a fix required somewhere here?

zacharybrown204 replied:
Comment removed

akuny commented on web-client/src/views/DocketRecord/FilingsAndProceedings.tsx:
```diff
@@ -143,6 +156,39 @@ export const FilingsAndProceedings = connect<
+                      link
+                      className={classNames(
+                        'text-right',
+                        // entry.isStricken && 'stricken-docket-record', // NOTE (#8546): This may be unnecessary
```
Is this comment necessary?

zacharybrown204 replied:
Removed

akuny commented on web-client/src/presenter/computeds/addCourtIssuedDocketEntryHelper.ts:
```diff
@@ -77,6 +99,8 @@ export const addCourtIssuedDocketEntryHelper = (
   return {
     documentTypes,
     formattedDocumentTitle,
+    caseMotions,
```
Is it worth updating the unit tests for this computed to cover these two new things?

jimlerza committed:
Merge branch 'test' into 8546-connect-order-to-motion-revamped-to-test-1761846877

akuny commented on cypress/local-only/tests/integration/caseDetail/docketRecord/courtIssuedFiling/docket-clerk-applies-order-to-lead-case-motion.cy.ts:
```diff
@@ -0,0 +1,59 @@
+import { goToCase } from '../../../../../../helpers/caseDetail/go-to-case';
```
When this test is run multiple times locally, it creates new test motions and test orders each run, but every order is associated with the initially created motion (see screenshot below). Is this intentional? If not, could we update this test such that documents are associated as expected over the course of multiple runs?

<img width="889" height="668" alt="Screenshot 2025-10-31 at 10 28 52 AM" src="https://github.com/user-attachments/assets/a21879c5-b6ff-4126-8195-559950530d01" />

zacharybrown204 replied:
Updated test to be more idempotent. We will now capture the docket entry id of our new motion and select that one when creating a new order.

akuny reviewed (changes requested):
This looks good. Just a few areas where we might want to tighten up.

zacharybrown204 committed:
8546 PR updates

Removing misc comments
Using explicit imports from lodash
Extracting common code to helper function
Reworking cypress test to be more idempotent

zacharybrown204 committed:
8546 - Add test to cover updating docket entry with affected docket entries.

JayFlexy committed:
8546: add unit test coverage for case motions

JayFlexy committed:
Merge branch 'test' into 8546-connect-order-to-motion-revamped-to-test-1761846877

JayFlexy committed:
8546: rm comments

TomElliottFlexion commented on shared/src/business/entities/DocketEntry.ts:
```diff
@@ -186,6 +187,10 @@ export class DocketEntry extends JoiValidationEntity {
   public strickenBy?: string;
   public strickenByUserId?: string;
+  // public orderDocketEntryId?: string;
+  // public motionDisposition?: string; // 'GRANTED' | 'DENTIED' | 'PARTIAL';
```
Intentional?

TomElliottFlexion commented on shared/src/business/entities/EntityConstants.ts:
```diff
@@ -88,7 +87,17 @@ export const JURISDICTIONAL_OPTIONS = {
+export type DocketEntryRelation = {
+  disposition: string;
+  docketEntryId: string;
+  // docketEntryType: string;
```
Still needed?

JayFlexy replied:
Nope, good catch

TomElliottFlexion commented on shared/src/business/entities/JoiValidationEntity.ts:
```diff
@@ -104,7 +104,9 @@ export abstract class JoiValidationEntity {
     const errors = {};
     error.details.forEach(detail => {
-      if (!Number.isInteger(detail.context.key)) {
+      if (detail.path.length > 1) {
+        errors[join(detail.path, '-')] = detail.message;
+      } else if (!Number.isInteger(detail.context.key)) {
```
Nice!

TomElliottFlexion commented on shared/src/business/entities/courtIssuedDocument/CourtIssuedDocumentTypeH.ts:
```diff
@@ -1,14 +1,13 @@
   public attachments: boolean;
-  public documentTitle?: string;
+  public documentTitle: string;
   public documentType: string;
-  public eventCode?: string;
+  public eventCode: string;
```
Really appreciate these fixes

TomElliottFlexion commented on web-api/src/business/useCaseHelper/docketEntry/addAssociatedDocketEntries.ts:
```diff
@@ -0,0 +1,30 @@
+import { DocketEntry } from '@shared/business/entities/DocketEntry';
+import { upsertDocketEntryRelatedEntries } from '@web-api/persistence/postgres/docketEntries/upsertDocketEntryRelatedEntries';
+
+export async function addAssociatedDocketEntries(
+  casesToUpdate: Omit<RawCase, 'consolidatedCases'>[],
```
Perhaps a unit test for this one?

zacharybrown204 replied:
Added unit test for this helper

jimlerza committed:
Merge branch 'test' into 8546-connect-order-to-motion-revamped-to-test-1761846877

JayFlexy committed:
8546: address feedback, cleanup

zacharybrown204 committed:
8546 - Adding additional unit tests

JayFlexy reviewed (approved):
LGTM

zacharybrown204 committed:
Merge branch 'test' into 8546-connect-order-to-motion-revamped-to-test-1761846877

zacharybrown204 committed:
Merge branch 'test' into 8546-connect-order-to-motion-revamped-to-test-1761846877

zacharybrown204 committed:
8546 - Whitespace change to reshuffle flaking test

JayFlexy reviewed (approved):
🥇
